src/bundle/ui-dev/src/modules/common/draggable-dialog/draggable.dialog.js

```diff
@@ -0,0 +1,149 @@
+import React, { useRef, createContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { getRootDOMElement } from '@ibexa-admin-ui/src/bundle/Resources/public/js/scripts/helpers/context.helper';
+import { createCssClassNames } from '@ibexa-admin-ui/src/bundle/ui-dev/src/modules/common/helpers/css.class.names';
+
+export const DraggableContext = createContext();
+
+const DraggableDialog = ({ children, referenceElement, positionOffset }) => {
+    const rootDOMElement = getRootDOMElement();
+    const containerRef = useRef();
+    const dragOffsetPosition = useRef({ x: 0, y: 0 });
```

Comment:
Was the react state too slow or were there other problems with it? 🤔

Reply:
Other problems - this variable is used in drag handlers - it's recalculated (and also used) on mousemove event, which keeps state from moment when drag started, so too old for its purpose.